source/extensions/filters/network/zookeeper_proxy/zookeeper_utils.cc

```diff
@@ -0,0 +1,53 @@
+#include "extensions/filters/network/zookeeper_proxy/zookeeper_utils.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace ZooKeeperProxy {
+
+bool BufferHelper::peekInt32(Buffer::Instance& buffer, uint64_t& offset, int32_t& val) {
+  try {
+    val = buffer.peekBEInt<int32_t>(offset);
+    offset += sizeof(uint32_t);
+    return true;
+  } catch (EnvoyException& e) {
+    return false;
+  }
+}
+
+bool BufferHelper::peekBool(Buffer::Instance& buffer, uint64_t& offset, bool& val) {
+  if (buffer.length() < (offset + 1)) {
+    return false;
+  }
+
+  try {
+    std::string str;
+    str.assign(std::string(static_cast<char*>(buffer.linearize(1 + offset)), 1 + offset));
```

Comment:
Can you not use `peekInt` with size 1 for this?

Reply:
Nice, will do.